### PR TITLE
Defer registry lookup until after Tiltfile is loaded

### PR DESCRIPTION
**Pull request by nicksieger**
When only using docker compose as in #4882, looking up the registry is unnecessary and appears to block the UI until after the tiltfile is fully executed. 

- tiltfile: defer finding registry until needed
- vendor: make vendor updates

### CODE REVIEW

File: internal/tiltfile/tiltfile_state.go [hunk 1]
```diff
@@ -1000,12 +1001,23 @@ func (s *tiltfileState) k8sResourceForName(name string) (*k8sResource, error) {
 // Otherwise, we'll return the zero value of `s.defaultReg`, which is an empty registry.
 // It has side-effects (a log line) and so should only be called once.
 func (s *tiltfileState) decideRegistry() container.Registry {
-	if s.orchestrator() == model.OrchestratorK8s && !s.localRegistry.Empty() {
+	if s.orchestrator() != model.OrchestratorK8s {
+		return s.defaultReg
+	}
+
+	// Defer lookup of registry until now
+	if s.localRegistry == nil {
```
Comment (nicks) [inline internal/tiltfile/tiltfile_state.go:1009]: for what it's worth - i'm pretty sure LocalRegistry() is already memoized in the k8sClient itself, so no need to store it as a field on tiltfileState

**Review by nicks — APPROVED** [review]
LGTM!

File: internal/tiltfile/tiltfile_state.go (hunk 1 above)
Comment (nicksieger) [reply internal/tiltfile/tiltfile_state.go:1009]: Yeah, I wasn't sure whether it made sense to keep that struct field. I'm going to remove and clean up.

**Review by milas — APPROVED** [review]
🎉 🎉 🎉 🎉

**Comment by hookenz** [comment]
@nicksieger I have a local kubeconfig that is pointing to a cluster address that no longer exists (due to me roaming).
The 20s startup time has either re-appeared or is still happening.

**Pull request merged**